### PR TITLE
Generalize photosynthesis filenaming

### PR DESCRIPTION
**Pull request by wurDevTim**
**Describe your changes**
Phenovation changed the naming convention over time. Some small changes where made to add support for the new files. 
The new code is tested with data from both new and old cropreporters. 

**Type of update**
Is this an update of the cropreporter filenaming to make the OJIP analysis work with old and new cropreporters. 

**Associated issues**
https://github.com/danforthcenter/plantcv/issues/978

**Additional context**
The current git only contains files which uses the 'old' metadata format. 
I have some test images with the new format, who can send these to? Or where to upload? 
Maybe also add a test for these new files?

**For the reviewer**
See [this page](https://plantcv.readthedocs.io/en/stable/pr_review_process/) for instructions on how to review the pull request.
- [x] PR functionality reviewed in a Jupyter Notebook
- [x] All tests pass
- [x] Test coverage remains 100%
- [x] Documentation tested
- [x] New documentation pages added to `plantcv/mkdocs.yml`
- [x] Changes to function input/output signatures added to `updating.md`
- [x] Code reviewed
- [ ] PR approved

### CODE REVIEW

**Review by HaleySchuhl — DISMISSED** [review]
Hi @wurDevTim thanks for opening this PR. The code looks good, but the additional functionality at L78 - L80 and L121-L123 are not getting hit by the new test. Also a clarification sentence added to the documentation for `photosynthesis_read_cropreporter` would be good too!

**Comment by wurDevTim** [comment]
@HaleySchuhl, I see, to hit those lines I have to add testdata from our cropreporter (the new file format).
The current ones are called `PSII_PSD_supopt_temp_btx623_22_rep1.DAT` and `PSII_PSD_supopt_temp_btx623_22_rep1.INF`.
Is there a naming preference for the new files? To make it easy to keep the two version apart or can I use any name I like?

**Comment by HaleySchuhl** [comment]
@wurDevTim pretty much any name you'd like!

**Comment by wurDevTim** [comment]
@HaleySchuhl  Added the files by making subfolders for the different cropreporter versions. Also updated the docs, let me know if  there are any other things you like me to change.

**Comment by wurDevTim** [comment]
After quite some mails back and forth to phenovation we figured out that when 'SaveAllFrames' is set to '0' it only returns 3 frames: `Fdark, F0, Fm`  for the PSD and `Flight, Fp, Fmp` for the PSL. 

However,  the framelabels are not updated. therefore the code below tries to set frame labels to frames which don't exist:
```        
elif 'FqFmFrameFsp' in metadata:
   fsp_frame = int(metadata["FqFmFrameFsp"]) + 1
   fmp_frame = int(metadata["FqFmFrameFmp"]) + 1            
elif 'LtOjipFrameFsp' in metadata:
   fsp_frame = int(metadata["LtOjipFrameFsp"]) + 1
   fmp_frame = int(metadata["LtOjipFrameFmp"]) + 1
```

To make plantcv robust against this bug I changed the code to:
```
if metadata['SaveAllFrames'] == '0':
   fsp_frame = 1
   fmp_frame = 2
elif 'FqFmFrameFsp' in metadata:
   fsp_frame = int(metadata["FqFmFrameFsp"]) + 1
   fmp_frame = int(metadata["FqFmFrameFmp"]) + 1            
elif 'LtOjipFrameFsp' in metadata:
   fsp_frame = int(metadata["LtOjipFrameFsp"]) + 1
   fmp_frame = int(metadata["LtOjipFrameFmp"]) + 1
```

File: plantcv/plantcv/photosynthesis/read_cropreporter.py [hunk 1]
```diff
@@ -71,8 +71,17 @@ def _process_psd_data(ps, metadata):
...
+        # Phenovation does not update the framenumbers in the references.
+        if metadata['SaveAllFrames'] == '0':
+            f0_frame = 1
+            fm_frame = 2
+        elif 'FvFmFrameF0' in metadata:
+            f0_frame = int(metadata["FvFmFrameF0"]) + 1  
+            fm_frame = int(metadata["FvFmFrameFm"]) + 1
+        elif 'DkOjipFrameF0' in metadata:
+            f0_frame = int(metadata["DkOjipFrameF0"]) + 1  # data cube includes Fdark at the beginning
+            fm_frame = int(metadata["DkOjipFrameFm"]) + 1
```
Comment (nfahlgren) [inline plantcv/plantcv/photosynthesis/read_cropreporter.py:84]:
```suggestion
        # Default frames (when SaveAllFrames == 0)
        f0_frame = 1
        fm_frame = 2
        # If the method is labeled FvFm
        if 'FvFmFrameF0' in metadata:
            f0_frame = int(metadata["FvFmFrameF0"]) + 1
            fm_frame = int(metadata["FvFmFrameFm"]) + 1
        # If the method is labeled DkOjip
        if 'DkOjipFrameF0' in metadata:
            f0_frame = int(metadata["DkOjipFrameF0"]) + 1  # data cube includes Fdark at the beginning
            fm_frame = int(metadata["DkOjipFrameFm"]) + 1
```

File: plantcv/plantcv/photosynthesis/read_cropreporter.py [hunk 2]
```diff
@@ -109,8 +118,18 @@ def _process_psl_data(ps, metadata):
...
+        # Phenovation does not update the framenumbers in the references.
+        if metadata['SaveAllFrames'] == '0':
+            fsp_frame = 1
+            fmp_frame = 2
+        elif 'FqFmFrameFsp' in metadata:
+            fsp_frame = int(metadata["FqFmFrameFsp"]) + 1
+            fmp_frame = int(metadata["FqFmFrameFmp"]) + 1            
+        elif 'LtOjipFrameFsp' in metadata:
+            fsp_frame = int(metadata["LtOjipFrameFsp"]) + 1
+            fmp_frame = int(metadata["LtOjipFrameFmp"]) + 1
```
Comment (nfahlgren) [inline plantcv/plantcv/photosynthesis/read_cropreporter.py:132]:
```suggestion
        # Default frames (when SaveAllFrames == 0)
        fsp_frame = 1
        fmp_frame = 2
        # If the method is labeled FqFm
        if 'FqFmFrameFsp' in metadata:
            fsp_frame = int(metadata["FqFmFrameFsp"]) + 1
            fmp_frame = int(metadata["FqFmFrameFmp"]) + 1
        # If the method is labeled LtOjip
        if 'LtOjipFrameFsp' in metadata:
            fsp_frame = int(metadata["LtOjipFrameFsp"]) + 1
            fmp_frame = int(metadata["LtOjipFrameFmp"]) + 1
```

**Review by nfahlgren — CHANGES_REQUESTED** [review]
Hi @wurDevTim, sorry for the long delay. I think this looks great, I just have a few suggestions for simplifying the code a bit

**Review by nfahlgren — APPROVED** [review]

**Pull request merged**